plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/TreeViewerFilter.java

```diff
@@ -0,0 +1,55 @@
+package io.snyk.eclipse.plugin.views.snyktoolview;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jface.viewers.ILabelProvider;
+import org.eclipse.jface.viewers.TreeViewer;
+import org.eclipse.jface.viewers.Viewer;
+import org.eclipse.jface.viewers.ViewerFilter;
+
+public class TreeViewerFilter extends ViewerFilter {
+	private List<String> searchStrings;
```

Comment:
I think a better strategy for filtering is to use the scan cache directly
With the current behavior we have an issue when multiple filters are applied one after another
* Scan is executed now we are showing all issues.
* User decides to filter now the tree has filtered items
* User changes filter; now the new filter is based on the current tree results and not the cached issues.